types/merge-source-map.d.ts

```diff
@@ -1,17 +1,28 @@
-
+/**
+ * Ambient module declaration for the [merge-source-map library](https://github.com/keik/merge-source-map).
+ *
+ * At the time of this writing, no official nor third party type declaration files exist for v1.1.0 of library, and are
+ * recreated here for type safety purposes.
+ */
 declare module 'merge-source-map' {
-  interface SourceMap {
-    file: string;
-    mappings: string;
-    names: string[];
-    sourceRoot?: string;
-    sources: string[];
-    sourcesContent?: string[];
-    version: number;
-  }
-
-  namespace merge {}
-  function merge(oldMap: SourceMap, newMap: SourceMap): SourceMap;
+  /**
+   * Merge an existing sourcemap with a new one
+   *
+   * This type declaration differs from the output of running `tsc -d` on the entrypoint of the merge-source-map
+   * module. It prefers concrete `MergeSourceMap` formal arguments over `(object | string)`.
+   *
+   * @param oldMap the original sourcemap to merge
+   * @param newMap the new sourcemap to merge with the `oldMap`
+   * @returns the merged sourcemap, or undefined if both maps are falsy
+   */
+  function merge(
+    oldMap: import('../src/declarations').SourceMap,
```

Comment:
If there's a better way to do this, I'm very much in favor of it (even if reintroducing the type declaration duplication is the route we take)

Reply:
I'm curious how this affects the browser... seems unrelated but I've been bitten by this import syntax. Take a peek at the /test/browser-compile directory and if that doesn't have any errors, I'm good!

Reply:
Looks good! Gonna put up a separate PR to add a reminder to TAL at that when doing a release until we get testing underway there